### PR TITLE
hesiod 3.2.1 (new formula)

### PR DESCRIPTION
**Pull request by leee**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --new-formula <formula>` (after doing `brew install <formula>`)?

---

Signed-off-by: Eugene Lee leee@mit.edu

### CODE REVIEW

**Comment by leee** [comment]
The `version` stanza had to be parsed before `url` in order for `brew fetch --retry hesiod --build-bottle --force` to work. How terrible, `brew install` seems to be indifferent about the order...

**Comment by leee** [comment]
Hi maintainers,

I can't seem to please rubocop or your CI infrastructure here at all - `brew fetch --retry hesiod --build-bottle --force` fails if I have the `url` stanza before `version` because it attempts to download `https://github.com/achernya/hesiod/archive/hesiod-.tar.gz"` - without the version, and rubocop will complain just about this if I do.

Because it's quite clear that this is an issue elsewhere, I will provide a formula that disrespects rubocop by simply punting `version` until this can be fixed.

File: Formula/hesiod.rb
Comment (MikeMcQuaid) [inline Formula/hesiod.rb]: There are likely `=> :build` dependencies.

File: Formula/hesiod.rb
Comment (MikeMcQuaid) [inline Formula/hesiod.rb]: It's probably worth also checking f this is needed at runtime.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]
> Because it's quite clear that this is an issue elsewhere, I will provide a formula that disrespects rubocop by simply punting version until this can be fixed.

Apologies for the inconvenience. Your current version is looking good on the version front. 

A few final notes but almost there.

**Comment by leee** [comment]
Hi @MikeMcQuaid - thanks for the code review; checking for build/runtime dependencies completely fell out of my mind last night. Looking at the debian packaging, only `libidn` is needed at runtime.

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
@BrewTestBot test this please

**Comment by leee** [comment]
Hi @MikeMcQuaid - thanks for the approved CR.

I see the builds are not pleased when performing an audit:

```
Error Message

failed: brew audit hesiod --new-formula
Stacktrace

        ==> Installing or updating 'rubocop' gem
Successfully installed rainbow-2.1.0
Successfully installed ast-2.3.0
Successfully installed parser-2.3.1.4
Successfully installed powerpack-0.1.1
Successfully installed ruby-progressbar-1.8.1
Successfully installed unicode-display_width-1.1.1
Successfully installed rubocop-0.43.0
7 gems installed
Error: 1 problem in 1 formula
hesiod:
  * GitHub repository not notable enough (<20 forks, <20 watchers and <50 stars)
```

I did not encounter this when I performed a `brew audit`, but in any case... seeing as hesiod is provided as an NSS module in glibc, is in active use in a number of environments, most notably at @mit and at Iowa State University, and is a Debian package in the Archives, I see no reason why it should not be packaged for homebrew as well.

**Comment by ilovezfs** [comment]
@leee to see that error you'd need to run `brew audit --new-formula`

**Comment by MikeMcQuaid** [comment]
> I did not encounter this when I performed a brew audit, but in any case... seeing as hesiod is provided as an NSS module in glibc, is in active use in a number of environments, most notably at @mit and at Iowa State University, and is a Debian package in the Archives, I see no reason why it should not be packaged for homebrew as well.

I agree. This is a warning rather than a failure (we have no way to return a warning state for CI) so we'll include it regardless. Thanks for the justification though!

**Comment by MikeMcQuaid** [comment]
Thanks for your first contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference there's no need to sign-off your own commits. Thanks!

**Comment by leee** [comment]
@ilovezfs - unfortunately, PEBKAC does not apply here:

```
leee@null ~> brew audit hesiod --new-formula
leee@null ~> brew update
...
leee@null ~> brew audit hesiod --new-formula
hesiod:
  * GitHub repository not notable enough (<20 forks, <20 watchers and <50 stars)
Error: 1 problem in 1 formula
```

It looks like `brew fetch` hasn't been fixed yet, but hopefully it will be when I try to submit another Formula. Worst comes to worst, I can always go digging in `Homebrew/brew`, right? :^)

@MikeMcQuaid - thanks for your handholding on this pull request, couldn't have done it without you!

**Comment by MikeMcQuaid** [comment]
@leee Thanks for the kind words!

**Pull request closed without merging**
